mm2src/coins/nft.rs

```diff
@@ -624,12 +680,61 @@ async fn get_moralis_nft_list(
             } else {
                 break;
             }
+        } else {
+            break;
         }
     }
     drop_mutability!(res_list);
     Ok(res_list)
 }
 
+pub(crate) async fn get_nfts_for_activation(
+    chain: &Chain,
+    my_address: &Address,
+    url: &Url,
+) -> MmResult<HashMap<String, NftInfo>, GetNftInfoError> {
+    let mut nfts_map = HashMap::new();
+    let uri_without_cursor = construct_moralis_uri_for_nft(url, &eth_addr_to_hex(my_address), chain)?;
+
+    // The cursor returned in the previous response (used for getting the next page).
+    let mut cursor = String::new();
+    loop {
+        let uri = format!("{}{}", uri_without_cursor, cursor);
+        let response = send_request_to_uri(uri.as_str()).await?;
+        if let Some(nfts_list) = response["result"].as_array() {
+            for nft_json in nfts_list {
+                let nft_moralis: NftFromMoralis = serde_json::from_str(&nft_json.to_string())?;
```

Comment:
I think you can use `NftFromMoralis::deserialize(nft_json)` here directly without the needless conversation to string in every loop.
```rust
 let nft_moralis = NftFromMoralis::deserialize(nft_json)?;
```

Reply:
this loop `for nft_json in nfts_list` can be abstracted into another function I guess

Reply:
> I think you can use NftFromMoralis::deserialize(nft_json)

Thanks, did it

> the inner loop for nft_json in nfts_list can be abstracted into another function I guess


Implemented `process_nft_list_for_activation` and `process_transfer_list` functions with for loops, now `get_nfts_for_activation` and `get_moralis_nft_transfers` function look much better. Thanks!

I decided not to move for loop from `get_moralis_nft_list` function, as it already utilizes inner functions in loop to make it more readable.